src/main.rs

```diff
@@ -34,6 +46,327 @@ pub mod test_harness;
+    // swap is finalized.
+    SwapFinalized(LocalSwapId, RemoteData),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Action {
+    PublishNewOrder(Order),
+    StartSwapExecution(LocalSwapId, RemoteData),
+}
+
+// Bundles the state of the application
+pub struct Maker {
+    btc_balance: bitcoin::Amount,
+    dai_balance: dai::Amount,
+    btc_fee: bitcoin::Amount,
+    dai_fee: dai::Amount,
+    btc_locked_funds: bitcoin::Amount,
+    dai_locked_funds: dai::Amount,
+    btc_max_sell_amount: bitcoin::Amount,
+    dai_max_sell_amount: dai::Amount,
+    rate: MidMarketRate,
+    spread: Spread,
+
+    active_orders: u32,
+}
+
+impl Maker {
+    // TODO: Proper constructor
+    pub fn new(intial_bicoin_balance: crate::bitcoin::Amount, initial_rate: MidMarketRate) -> Self {
+        let ZERO_DAI = dai::Amount::from_dai_trunc(0.0).unwrap();
+        let ZERO_BTC = crate::bitcoin::Amount::from_btc(0.0).unwrap();
+
+        Maker {
+            btc_balance: intial_bicoin_balance,
+            dai_balance: ZERO_DAI.clone(),
+            btc_fee: ZERO_BTC,
+            dai_fee: ZERO_DAI.clone(),
+            btc_locked_funds: ZERO_BTC,
+            dai_locked_funds: ZERO_DAI.clone(),
+            btc_max_sell_amount: ZERO_BTC,
+            dai_max_sell_amount: ZERO_DAI,
+            rate: initial_rate,
+            // 300 ^= 3.00%
+            spread: Spread::new(300).expect("default spread works"),
+            active_orders: 0,
+        }
+    }
+
+    pub fn handle_event(&mut self, event: NetworkEvent) -> Option<Action> {
+        match event {
+            NetworkEvent::OrderExpired(order) => {
+                // TODO: Bookkeeping, decisionmaking if new order
+
+                // TODO: creating a new order should not take the wallet / book
+                // let new_order = new_dai_bitcoin_order();
+                let new_order = Order {
+                    expiration_timestamp: Utc::now(),
+                };
+
+                Some(Action::PublishNewOrder(order))
+            }
+            NetworkEvent::OrderTakeRequest(order) => {
+                // TODO: Decisionmaking if we actually allow taking the order
+
+                // TODO: Action that indicates if we accepted or declined the order to be emitted, so we can react in event loop
+
+                unimplemented!()
+            }
+            NetworkEvent::SwapFinalized(local_swap_id, remote_date) => {
+                // TODO: Bookkeeping, lock up funds
+                Some(Action::StartSwapExecution(local_swap_id, remote_date))
+            }
+            _ => None,
+        }
+    }
+
+    //
+    pub fn update_rate(&mut self, new_rate: ()) {}
+
+    // the balance is to be updated once the trade was actually setteled, i.e. the swap execution is finished
+    pub fn update_bitcoin_balance(&mut self, balance: bitcoin::Amount) {
+        self.btc_balance = balance;
+    }
+
+    pub fn track_failed_rate(&mut self, error: anyhow::Error) {}
```

Comment:
I think it will depend on if the `Maker` wants to use this information as part of his decision-making. If so, I'd say it should remain in the maker as part of its state. 

@D4nte we can think this through tomorrow and take a decision.